blockingconcurrentqueue.h

```diff
@@ -7,412 +7,16 @@
 #pragma once
 
 #include "concurrentqueue.h"
+#include "semaphore.hpp"
```

Comment:
This should be named `semaphore.h` for consistency, or better yet `lightweightsemaphore.h` since that's what it is.